File: 0074_Search a 2D Matrix/solution.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        for (int i = 0; i < matrix.size(); i++) {
            int left = 0;
            int right = matrix[i].size() - 1;
            while (left <= right) {
                if (matrix[i][left] != target && matrix[i][right] != target) {
                    left++;
                    right--;
                }
                else {
                    return true;
                }
            }
        }
        return false;
    }
};

File: 0074_Search a 2D Matrix/solution.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for i in range(len(matrix)):
            left = 0
            right = len(matrix[i]) - 1
            while left <= right:
                if matrix[i][left] != target and matrix[i][right] != target:
                    left += 1
                    right += -1
                else:
                    return True
        return False
                